### ISSUE TITLE
[Bug] Silent NaN corruption when serving VLM model without visual encoder weights

### ISSUE THREAD

**Issue by rouchenzi** [sgl-project/sglang#26745]
### Checklist

- [x] I searched related issues but found no solution.
- [x] The bug persists in the latest version.
- [x] Issues without environment info and a minimal reproducible demo are hard to resolve and may receive no feedback.
- [x] If this is not a bug report but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [x] Please use English. Otherwise, it will be closed.

### Bug Description

When serving a VLM model (e.g., Qwen3.5-122B-A10B) with a checkpoint that's missing visual encoder weights (e.g., after Megatron training that only trains the language model), SGLang silently serves NaN/garbage output for all chat requests. No error or warning is raised at any point.

### Root Cause
3 issues combine:

1. **No weight loading validation:** `load_weights()` doesn't check or report which parameters remain uninitialized. Visual encoder params are allocated with `torch.empty()` and never loaded from checkpoint — they contain arbitrary memory. No warning is emitted.

2. **Warmup sends image request to uninitialized visual encoder:** `_execute_server_warmup()` detects VLM architecture and sends a 32×32 PNG. The uninitialized visual encoder produces NaN embeddings. GDN (recurrent) layers propagate NaN to subsequent positions. Full attention layers write NaN K,V to the KV pool. After warmup completes, these NaN KV slots are cached in the radix tree. Subsequent chat requests that share a prefix with warmup (`<|im_start|>user\n`) pick up these corrupted slots via `cache_unfinished_req`'s prefix slot reuse — even though the scheduler correctly rejected prefix sharing (`prefix_len=0`) and extend computed correct KV with fresh slots. Once corrupted KV enters the cache, it silently propagates to all subsequent requests sharing that prefix.

### Impact
- ALL /v1/chat/completions requests produce garbage after warmup
- /v1/completions (raw tokens, no chat template) is unaffected (no shared prefix)
- No error/warning in logs caused silent failure

#### 1. Weight loading validation (aligns with RFC #24703)

 After load_weights(), compare loaded keys against model.state_dict().keys(). Warn or error on unloaded parameters:
 ```
  loaded_keys = model.load_weights(weights)
  all_keys = set(model.state_dict().keys())
  unloaded = all_keys - loaded_keys
  if unloaded:
      raise ValueError(f"Weights not initialized from checkpoint: {unloaded}")
 ```

#### 2. Warmup should validate visual encoder state

 Before sending image warmup, check if visual weights are actually loaded:
 ```
  if hasattr(model, 'visual') and model.visual is not None:
      first_param = next(model.visual.parameters())
      if first_param.isnan().any() or first_param.abs().max() > 1e30:
          # Skip image warmup, use text-only
          pass
 ```

#### 3. A general "skip vision encoder" mode

 We note that SGLang's --language-only flag is specifically designed for encoder-decoder prefill disaggregation (EPD), not a general "skip  vision encoder" mode (see #21140 > "language_only requires --encoder-urls and is restricted to Qwen architectures only").

 vLLM's --language-model-only covers this use case cleanly. Would SGLang consider a similar general-purpose flag (separate from EPD's --language-only) that:
 - Skips visual module allocation (or stubs it out)
 - Skips image warmup
 - Rejects multimodal requests at the API level

If the current architecture doesn't support this cleanly, fixes #1 and #2 would already prevent the silent failure

### Related Issues
  - #24703 — RFC: Refactor weight loading code (proposes set[str] return from load_weights)
  - #22701 — Corrupted weights, missing quant init for multimodal models
  - #20043 — NaN in hidden states with modelopt_fp4 (--enable-nan-detection)
  - #22935 — Mamba radix cache: fresh prefill prefix hits become 0-hit after split
  - #21140 — Clarification that --language-only is for EPD, not general "skip vision encoder"

### Reproduction

  1. Serve a VLM model with checkpoint missing visual weights
  2. Server warmup fires automatically (sends image)
  3. Send a chat request: curl /v1/chat/completions -d '{"messages":[{"role":"user","content":"What is 2+2?"}]}'
  4. Observe garbage output > no error in logs

### Environment

- Model: Qwen3.5-122B-A10B (hybrid GDN + attention, VLM)
- Checkpoint: Megatron-converted, 333 visual encoder params missing

python3 -m sglang.check_env
```
Python: 3.12.3
CUDA available: True
GPU 0-7: NVIDIA H200 (Compute Capability 9.0)
CUDA: 12.9 / Driver: 580.126.09
PyTorch: 2.9.1+cu129
sglang: 0.5.9
flashinfer: 0.6.3
triton: 3.5.1
transformers: 5.2.0
Topology: 8x H200, NV18 interconnect
```
